api/envoy/config/route/v3/route_components.proto

```diff
@@ -542,6 +542,19 @@ message RouteMatch {
     //
     // Expect the value to not contain ``?`` or ``#`` and not to end in ``/``
     string path_separated_prefix = 14 [(validate.rules).string = {pattern: "^[^?#]+[^?#/]$"}];
+
+    // If specified, the route is a template match rule meaning that the
+    // ``:path`` header (without the query string) must match the given
+    // ``path_template_match`` pattern.
+    //
+    // For example:
+    //
+    // * ``/videos/*/*/*.m4s`` would match `videos/123414/hls/1080p5000_00001.m4s``
+    //
+    // * ``/videos/{file}`` would match ``/videos/1080p5000_00001.m4s"
+    //
+    // * ``/**.mpd`` would match ``/content/123/india/dash/55/manifest.mpd``
+    string path_template_match = 15 [(validate.rules).string = {min_len: 1}];
```

Comment:
ISTM path_pattern_match would be a more consistent name than path_template_match given the other code changed in this PR. Are you using "template" to be consistent with terminology that already exists somewhere?

Reply:
Yes it matches with the Cloud Media config 
(https://cloud.google.com/media-cdn/docs/routing#path-matching)

Reply:
Is this also the path template matching also used in gRPC transcoding? https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#httprule

Reply:
Yes very similar markup and pattern rules it looks like